### PR TITLE
feat(cloudtrail): cloudtrail service and checks

### PR DESCRIPTION
**Pull request by n4ch04**
### Context 

We need to migrate cloudtrail checks from version 2 to version 3

### Description

Includes:
- cloudtrail service
- checks
     - cloudtrail_cloudwatch_logging_enabled
     - cloudtrail_kms_encryption_enabled
     - cloudtrail_log_file_validation_enabled
     - cloudtrail_logs_s3_bucket_access_logging_enabled
     - cloudtrail_logs_s3_bucket_is_not_publicly_accessible
     - cloudtrail_multi_region_enabled
  - Changes in s3 service

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: providers/aws/services/cloudtrail/cloudtrail_cloudwatch_logging_enabled/cloudtrail_cloudwatch_logging_enabled.metadata.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    "Severity": "high",
+    "ResourceType": "AwsCloudTrailTrail",
+    "Description": "Ensure CloudTrail trails are integrated with CloudWatch Logs",
+    "Risk": "Sending CloudTrail logs to CloudWatch Logs will facilitate real-time and historic activity logging based on user; API; resource; and IP address; and provides opportunity to establish alarms and notifications for anomalous or sensitivity account activity.",
+    "RelatedUrl": "",
+    "Remediation": {
+      "Code": {
+        "CLI": "aws cloudtrail update-trail --name <trail_name> --cloudwatch-logs-log-group- arn <cloudtrail_log_group_arn> --cloudwatch-logs-role-arn <cloudtrail_cloudwatchLogs_role_arn>",
+        "NativeIaC": "",
+        "Other": "",
```
Comment (MrCloudSec) [inline providers/aws/services/cloudtrail/cloudtrail_cloudwatch_logging_enabled/cloudtrail_cloudwatch_logging_enabled.metadata.json]:
```suggestion
        "Other": "https://docs.bridgecrew.io/docs/logging_4#aws-console",
```

File: providers/aws/services/cloudtrail/cloudtrail_log_file_validation_enabled/cloudtrail_log_file_validation_enabled_test.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+            assert len(result) == 1
+            assert result[0].status == "FAIL"
+            assert search("log file validation disabled", result[0].status_extended)
+            assert result[0].resource_id == trail_name_us
+            assert result[0].resource_arn == trail_us["TrailARN"]
+
+    @mock_cloudtrail
+    @mock_s3
+    def test_various_trails_with_and_without_logging_enabled(self):
```
Comment (MrCloudSec) [inline providers/aws/services/cloudtrail/cloudtrail_log_file_validation_enabled/cloudtrail_log_file_validation_enabled_test.py]:
```suggestion
    def test_various_trails_with_and_without_logging_validation(self):
```

File: providers/aws/services/cloudtrail/cloudtrail_log_file_validation_enabled/cloudtrail_log_file_validation_enabled_test.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+from unittest import mock
+
+from boto3 import client
+from moto import mock_cloudtrail, mock_s3
+
+
+class Test_cloudtrail_log_file_validation_enabled:
+    @mock_cloudtrail
+    @mock_s3
+    def test_no_logging(self):
```
Comment (MrCloudSec) [inline providers/aws/services/cloudtrail/cloudtrail_log_file_validation_enabled/cloudtrail_log_file_validation_enabled_test.py]:
```suggestion
    def test_no_logging_validation(self):
```

File: providers/aws/services/cloudtrail/cloudtrail_logs_s3_bucket_access_logging_enabled/cloudtrail_logs_s3_bucket_access_logging_enabled.metadata.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "Severity": "critical",
+    "ResourceType": "AwsCloudTrailTrail",
+    "Description": "Ensure S3 bucket access logging is enabled on the CloudTrail S3 bucket",
+    "Risk": "Server access logs can assist you in security and access audits; help you learn about your customer base; and understand your Amazon S3 bill.",
+    "RelatedUrl": "",
+    "Remediation": {
+      "Code": {
+        "CLI": "",
+        "NativeIaC": "",
+        "Other": "",
```
Comment (MrCloudSec) [inline providers/aws/services/cloudtrail/cloudtrail_logs_s3_bucket_access_logging_enabled/cloudtrail_logs_s3_bucket_access_logging_enabled.metadata.json]:
```suggestion
        "Other": "https://docs.bridgecrew.io/docs/logging_6#aws-console",
```

File: providers/aws/services/cloudtrail/cloudtrail_logs_s3_bucket_is_not_publicly_accessible/cloudtrail_logs_s3_bucket_is_not_publicly_accessible.metadata.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+    "Provider": "aws",
+    "CheckID": "cloudtrail_logs_s3_bucket_is_not_publicly_accessible",
+    "CheckTitle": "Ensure the S3 bucket CloudTrail logs to is not publicly accessible",
```
Comment (MrCloudSec) [inline providers/aws/services/cloudtrail/cloudtrail_logs_s3_bucket_is_not_publicly_accessible/cloudtrail_logs_s3_bucket_is_not_publicly_accessible.metadata.json]:
```suggestion
    "CheckTitle": "Ensure the S3 bucket CloudTrail logs is not publicly accessible",
```

File: providers/aws/services/cloudtrail/cloudtrail_logs_s3_bucket_is_not_publicly_accessible/cloudtrail_logs_s3_bucket_is_not_publicly_accessible.metadata.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    "Severity": "critical",
+    "ResourceType": "AwsCloudTrailTrail",
+    "Description": "Ensure the S3 bucket CloudTrail logs to is not publicly accessible",
+    "Risk": "Allowing public access to CloudTrail log content may aid an adversary in identifying weaknesses in the affected accounts use or configuration.",
+    "RelatedUrl": "",
+    "Remediation": {
+      "Code": {
+        "CLI": "",
+        "NativeIaC": "",
+        "Other": "",
```
Comment (MrCloudSec) [inline providers/aws/services/cloudtrail/cloudtrail_logs_s3_bucket_is_not_publicly_accessible/cloudtrail_logs_s3_bucket_is_not_publicly_accessible.metadata.json]:
```suggestion
        "Other": "https://docs.bridgecrew.io/docs/logging_3#aws-console",
```

File: providers/aws/services/cloudtrail/cloudtrail_multi_region_enabled/cloudtrail_multi_region_enabled.metadata.json [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    "ResourceType": "AwsCloudTrailTrail",
+    "Description": "Ensure CloudTrail is enabled in all regions",
+    "Risk": "AWS CloudTrail is a web service that records AWS API calls for your account and delivers log files to you. The recorded information includes the identity of the API caller; the time of the API call; the source IP address of the API caller; the request parameters; and the response elements returned by the AWS service.",
+    "RelatedUrl": "",
+    "Remediation": {
+      "Code": {
+        "CLI": "",
+        "NativeIaC": "",
+        "Other": "",
+        "Terraform": ""
```
Comment (MrCloudSec) [inline providers/aws/services/cloudtrail/cloudtrail_multi_region_enabled/cloudtrail_multi_region_enabled.metadata.json]:
```suggestion
        "CLI": "aws cloudtrail create-trail --name <trail_name> --bucket-name <s3_bucket_for_cloudtrail> --is-multi-region-trail aws cloudtrail update-trail --name <trail_name> --is-multi-region-trail ",
        "NativeIaC": "https://docs.bridgecrew.io/docs/logging_1#cloudformation",
        "Other": "https://docs.bridgecrew.io/docs/logging_1#aws-console",
        "Terraform": "https://docs.bridgecrew.io/docs/logging_1#terraform"
```

File: providers/aws/services/s3/s3_service.py [hunk 8]
```diff
@@ -86,9 +114,27 @@ class Bucket:
     versioning: bool
     logging: bool
     region: str
+    acl_grantee: list
```
Comment (MrCloudSec) [inline providers/aws/services/s3/s3_service.py]:
```suggestion
    acl_grantee: list [ACL_Grantee]
```

File: providers/aws/services/cloudtrail/cloudtrail_logs_s3_bucket_access_logging_enabled/cloudtrail_logs_s3_bucket_access_logging_enabled.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+                report.region = trail.region
+                report.resource_id = trail.name
+                report.resource_arn = trail.trail_arn
+                report.status = "FAIL"
+                if trail.is_multiregion:
+                    report.status_extended = f"Multiregion Trail {trail.name} S3 bucket access logging is not enabled for bucket {trail_bucket}"
+                else:
+                    report.status_extended = f"Single region Trail {trail.name} S3 bucket access logging is not enabled for bucket {trail_bucket}"
+                for bucket in s3_client.buckets:
+                    if trail_bucket == bucket.name and bucket.logging_target_bucket:
```
Comment (MrCloudSec) [inline providers/aws/services/cloudtrail/cloudtrail_logs_s3_bucket_access_logging_enabled/cloudtrail_logs_s3_bucket_access_logging_enabled.py]:
```suggestion
                    if trail_bucket == bucket.name and bucket.logging:
```

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Great job @n4ch04 🤩

**Comment by MrCloudSec** [comment]
Please, make sure that the severities in the metadata are correct.

**Comment by n4ch04** [comment]
Severity is corrected, ready to merge !!

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**